test/CMakeLists.txt

```diff
@@ -50,3 +50,38 @@ add_custom_target(installcheck
   PG_REGRESS=${PG_REGRESS}
   ${CMAKE_CURRENT_SOURCE_DIR}/pg_regress.sh ${PG_REGRESS_OPTS}
   USES_TERMINAL)
+
+set(PG_REGRESS_DIR "" CACHE PATH "Path to postgresql src regress directory.")
+
+if (NOT ${PG_REGRESS_DIR} STREQUAL "")
+
+set(BASE_TEST_INPUT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/base_test)
+set(BASE_TEST_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/base_test)
+set(BASE_TEST_CLUSTER ${BASE_TEST_OUTPUT_DIR}/testcluster)
+
+set(PG_REGRESS_OPTS_BASE
+  --temp-instance=${TEST_CLUSTER}
+  --temp-config=${TEST_OUTPUT_DIR}/postgresql.conf
+  --host=${TEST_PGHOST}
+  --port=${TEST_PGPORT}
+  --user=${TEST_PGUSER}
+  --load-language=plpgsql
+  --inputdir=${BASE_TEST_INPUT_DIR}
+  --outputdir=${BASE_TEST_OUTPUT_DIR}
+  --load-extension=timescaledb
+  --dlpath=${PROJECT_BINARY_DIR}/src)
+
+file(MAKE_DIRECTORY ${BASE_TEST_OUTPUT_DIR})
+file(WRITE ${BASE_TEST_OUTPUT_DIR}/postgresql.conf "shared_preload_libraries=timescaledb")
```

Comment:
Can't we reuse the file already generated for regular tests? Indeed, that seems to be what your are doing in PG_REGRESS_OPTS_BASE anyway so I don't see the point of creating this extra file.

Reply:
sure